packages/nuxt/src/app/plugins/view-transitions-safe.client.ts

```diff
@@ -0,0 +1,8 @@
+import { defineNuxtPlugin } from '#app/nuxt'
+import plugin from '#app/plugins/view-transitions.client'
+
+export default defineNuxtPlugin((nuxtApp) => {
+  if (!document.startViewTransition || window.matchMedia('(prefers-reduced-motion: reduce)').matches) { return }
```

Comment:
Making a separate safe plugin seemed like the best way to do this? Plugins don't seem to have any access to the nuxt config option?

Reply:
We can specifically expose it via this template:

https://github.com/nuxt/nuxt/blob/09161d005da48b0ba82308093e509a53d9a8b899/packages/nuxt/src/core/templates.ts#L367-L378

... and then import the value from '#build/nuxt.config.mjs' as we do here:

https://github.com/nuxt/nuxt/blob/d7b11164434e5ae318cd8cf85417362fddb99d71/packages/nuxt/src/app/plugins/payload.client.ts#L7-L8

Reply:
(Alternatively, the approach you've taken here would work too.)

Reply:
Is there a preference for one or the other I'm happy to change it?

Reply:
Either way works for me 👍

Reply:
Okay in that case this should be ready to go when the team are :)

Reply:
actually, let me backpedal on this.

the reason is that we're doing more build-time optimisations by parsing plugins, and importing and running a plugin might cause issues in future.

if we have one plugin and read the config option from `#build/nuxt.config.mjs` that would be better

Reply:
No worries I'll get that changed!

Reply:
I think this is done correctly now :)